### PR TITLE
Features/validation frontend test

### PR DESCRIPTION
**Pull request by Mikanebu**
* added sample datasets to cover use cases
* added tests to test small dataset with multiple resources and duplicated row
* added tests to test big dataset with invalid metadata and data itself
* added tests to test processing dataset by providing invalid `flow.yml`
* added tests for testing all specstore statuses for sample datasets
* updated `travis.yml` by adding extra commands to push sample datasets
* updated `env.template`, moved some env.variables inside script
* updated `statuses.test.js` by removing unused dataset from the list of object

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,16 @@ install:
 - npm install
 - npm install -g git+https://github.com/datahq/datahub-cli.git
 before_script:
-- "data push test/fixtures/validate-redirection-dataset --api=$API"
-- sleep 90
+- "data push test/fixtures/validate-redirection-dataset --private --api=$API"
+- sleep 60
```
Comment (zelima) [inline .travis.yml:14]: why are you sleeping after every push? push all an sleep once at the end, wdyt?

File: scripts/index.js [hunk 2]
```diff
@@ -159,14 +157,28 @@ const frontendStatus = async (dataset,baseUrl,newLine,options) => {
...
     // tables
     const table = $('.htCore').find('tr').length
     if (table > 2) {
       statuses.tables = 'OK'
     } else {
       statuses.tables = 'NOT OK'
     }
+    // validation status report
+    const validationStatus = $('.alert-danger').find('h3').text()
+    if (validationStatus.length > 10) {
```
Comment (zelima) [inline scripts/index.js:169]: this is really strange. why 10, what are you testing here - the comment would be useful (in the code)

File: test/validation.test.js [hunk 3]
```diff
@@ -46,4 +63,105 @@ describe('dataset validation in frontend', function () {
...
+    const urlLatest = `${SPECSTORE}/${OWNERID}/${datasetsToTest[1].name}/latest`
+    let body = await apiStatus(urlLatest)
+    const revisionId = body.id.split('/').pop()
+    const revisionUrl = `${SPECSTORE}/${OWNERID}/${datasetsToTest[1].name}/${revisionId}`
+    body = await apiStatus(revisionUrl)
+    expect(body.spec_contents.meta.dataset).to.equal(`${datasetsToTest[1].name}`)
+    expect(body.state).to.equal('FAILED')
+  })
+  it(`small invalid dataset with duplicated row on ${DOMAIN}`, async function () {
+    this.timeout(1800000)
```
Comment (zelima) [inline test/validation.test.js:77]: Waw half hour timeout?

**Review by zelima — CHANGES_REQUESTED** [review]
@Mikanebu It is very hard to follow what exactly you are testing. Can you list exactly what cases are you testing in a human-readable way? Eg are you testing list of validation errors against invalid data you have Eg. 
* 3rd row is missing column in data - are you testing that validation report says that 
* header is missing - does report say that? 
* Eg duplicate row - does validation report say duplicate row. 
* also please list what kind invalid data you have,  (maybe in README?)

File: scripts/index.js (hunk 2 above)
Comment (Mikanebu) [reply scripts/index.js:169]: it is validation error message, I cannot create specific validation for each case. I generalized for all tests.

File: test/validation.test.js (hunk 3 above)
Comment (Mikanebu) [reply test/validation.test.js:77]: It is temporary, it will be removed when I refactor to `ava`.

**Comment by Mikanebu** [comment]
@zelima Added use cases into README.md

File: scripts/index.js (hunk 2 above)
Comment (zelima) [reply scripts/index.js:169]: @Mikanebu but what is the point of creating invalid data fail with all kind of invalidity then? it would fail in case of only one of them, right?
In this case I was requesting just to make comment on line 168 probably
```
// checking xxx
```

Comment (zelima) [reply scripts/index.js:169]: But we need to create specific validation for each case I think. That's all these tests are about, no?

**Comment by zelima** [comment]
@Mikanebu Let's merge this for now and create a new issue about this:
>we need to create specific validation for each case I think
